paasta_tools/deployd/common.py

```diff
@@ -1,31 +1,61 @@
 import logging
 import time
 from collections import namedtuple
+from queue import PriorityQueue
 from queue import Queue
 from threading import Thread
 
 from paasta_tools.marathon_tools import DEFAULT_SOA_DIR
 from paasta_tools.marathon_tools import get_all_marathon_apps
 from paasta_tools.marathon_tools import get_marathon_client
 from paasta_tools.marathon_tools import load_marathon_config
+from paasta_tools.marathon_tools import load_marathon_service_config
 from paasta_tools.marathon_tools import load_marathon_service_config_no_cache
 from paasta_tools.utils import InvalidJobNameError
 from paasta_tools.utils import NoDeploymentsAvailable
 from paasta_tools.utils import NoDockerImageError
 
 BounceTimers = namedtuple('BounceTimers', ['processed_by_worker', 'setup_marathon', 'bounce_length'])
-ServiceInstance = namedtuple(
+ServiceInstanceTuple = namedtuple(
     'ServiceInstance', [
         'service',
         'instance',
         'bounce_by',
         'watcher',
         'bounce_timers',
         'failures',
+        'priority',
     ],
 )
 
 
+def ServiceInstance(service, instance, watcher, cluster, bounce_by, failures=0, bounce_timers=None, priority=None):
```

Comment:
My first instinct was for a sub class. I got a bit scared when I read the `namedtuple` source :-S

Let me take another look though, I agree that it would be the right way to do it